### DISCUSSION TITLE
Tough Basic Concepts

### DISCUSSION THREAD

**Discussion by Leoveth** [GDQuest/learn-gdscript#495] [Ideas]
Hi, I'm loving the course and, having practically no previous programming background, it has helped me solidify many concepts and good practices.
I would kindly ask you to add explanations for export (keyword), class and type. Those are keywords and concepts that are used here and there and it would be nice to know in simple words what they mean. I feel that it's best to learn this at an early stage, just like the extend keyword was explained, that felt really helpful to me.

**Comment by NathanLovato** [comment]
We introduce the notion of type at the end of this course already. The export keyword and classes are difficult to cover here because:

1. Export only affects the Godot editor. This app isn't the Godot editor, so we can't show the keyword in action.
2. Classes in Godot also tie into nodes that you mostly manipulate through the Godot editor. We'll explain them in [learn to code from zero with Godot](https://gdquest.mavenseed.com/courses/learn-to-code-from-zero-with-godot) but not in this app.
